pallets/ddc-clusters/src/migrations.rs

```diff
@@ -402,3 +419,185 @@ pub mod v2 {
 		}
 	}
 }
+
+pub mod v3 {
+	use frame_support::{
+		pallet_prelude::*,
+		traits::{Get, OnRuntimeUpgrade},
+		weights::Weight,
+	};
+	use sp_std::marker::PhantomData;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	use super::*;
+	pub fn migrate_to_v3<T: Config>() -> Weight {
+		let on_chain_version = Pallet::<T>::on_chain_storage_version();
+		let current_version = Pallet::<T>::current_storage_version();
+
+		info!(
+			target: LOG_TARGET,
+			"Running migration with current storage version {:?} / onchain {:?}",
+			current_version,
+			on_chain_version
+		);
+
+		if on_chain_version == 2 && current_version == 3 {
+			let count = Clusters::<T>::iter().count() as u64;
+			info!(
+				target: LOG_TARGET,
+				" >>> Updating DDC Cluster storage. Migrating {} clusters...", count
+			);
+
+			Clusters::<T>::translate::<v1::Cluster<T::AccountId>, _>(
+				|cluster_id: ClusterId, cluster: v1::Cluster<T::AccountId>| {
+					info!(target: LOG_TARGET, "     Migrating cluster for cluster ID {:?}...", cluster_id);
+
+					Some(Cluster {
+						cluster_id: cluster.cluster_id,
+						manager_id: cluster.manager_id,
+						reserve_id: cluster.reserve_id,
+						props: cluster.props,
+						status: cluster.status,
+						last_validated_era_id: 0,
+					})
+				},
+			);
+
+			// Update storage version.
+			StorageVersion::new(3).put::<Pallet<T>>();
+			info!(
+				target: LOG_TARGET,
+				"Upgraded {} records, storage to version {:?}",
+				count,
+				current_version
+			);
+
+			T::DbWeight::get().reads_writes(count + 2, count + 1)
+		} else {
+			info!(target: LOG_TARGET, " >>> Unused migration!");
+			T::DbWeight::get().reads(1)
+		}
+	}
+
+	pub struct MigrateToV3<T>(PhantomData<T>);
+
+	impl<T: Config> OnRuntimeUpgrade for MigrateToV3<T> {
+		fn on_runtime_upgrade() -> Weight {
+			migrate_to_v3::<T>()
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::DispatchError> {
+			let prev_count = Clusters::<T>::iter().count();
+
+			Ok((prev_count as u64).encode())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(prev_state: Vec<u8>) -> Result<(), sp_runtime::DispatchError> {
+			let prev_count: u64 = Decode::decode(&mut &prev_state[..])
+				.expect("pre_upgrade provides a valid state; qed");
+
+			let post_count = Clusters::<T>::iter().count() as u64;
+			ensure!(
+				prev_count == post_count,
+				"the cluster count before and after the migration should be the same"
+			);
+
+			let current_version = Pallet::<T>::current_storage_version();
+			let on_chain_version = Pallet::<T>::on_chain_storage_version();
+
+			ensure!(current_version == 3, "must_upgrade");
```

Comment:
In `post_upgrade` you can additionally check what your migrations modified, i.e. you can check that `last_validated_era_id` field is set to 0 for all clusters and there is no clusters with other values.